src/checkbox/tests/unit/Checkbox.ts

```diff
@@ -1,151 +1,238 @@
+	beforeEach() {
+		widget = harness(Checkbox);
 	},
 
-	'default node attributes'() {
-		const checkbox = new Checkbox();
-		const vnode = <VNode> checkbox.__render__();
-		const inputNode = vnode.children![0].children![0];
+	afterEach() {
+		widget.destroy();
+	},
 
-		assert.strictEqual(inputNode.vnodeSelector, 'input');
-		assert.strictEqual(inputNode.properties!.type, 'checkbox');
-		assert.strictEqual(inputNode.properties!.checked, false);
+	simple() {
+		widget.expectRender(v('div', {
+			classes: widget.classes(css.root)
+		}, [ v('div', {
+				classes: widget.classes(css.inputWrapper)
+			}, [
+				null,
+				null,
+				v('input', {
+					'aria-describedby': undefined,
+					'aria-invalid': null,
+					'aria-readonly': null,
+					bind: true,
+					checked: false,
+					classes: widget.classes(css.input),
+					disabled: undefined,
+					name: undefined,
+					onblur: widget.listener,
+					onchange: widget.listener,
+					onclick: widget.listener,
+					onfocus: widget.listener,
+					onmousedown: widget.listener,
+					onmouseup: widget.listener,
+					ontouchstart: widget.listener,
+					ontouchend: widget.listener,
+					ontouchcancel: widget.listener,
+					readOnly: undefined,
+					required: undefined,
+					type: 'checkbox',
+					value: undefined
+				})
+			])
+		]));
 	},
 
-	'correct node attributes'() {
-		const checkbox = new Checkbox();
-		checkbox.__setProperties__({
+	'properties and attributes'() {
+		const checkboxProperties: CheckboxProperties = {
 			checked: true,
 			describedBy: 'id1',
 			disabled: true,
 			formId: 'id2',
 			invalid: true,
 			label: 'foo',
+			mode: Mode.toggle,
 			name: 'bar',
+			offLabel: 'Off',
+			onLabel: 'On',
 			readOnly: true,
 			required: true,
 			value: 'qux'
-		});
-		const vnode = <VNode> checkbox.__render__();
-		const labelNode = vnode.children![0];
-		const inputNode = vnode.children![1].children![0];
-
-		assert.isTrue(inputNode.properties!.checked);
-		assert.strictEqual(inputNode.properties!['aria-describedby'], 'id1');
-		assert.isTrue(inputNode.properties!.disabled);
-		assert.strictEqual(inputNode.properties!['aria-invalid'], 'true');
-		assert.strictEqual(inputNode.properties!.name, 'bar');
-		assert.isTrue(inputNode.properties!.readOnly);
-		assert.strictEqual(inputNode.properties!['aria-readonly'], 'true');
-		assert.isTrue(inputNode.properties!.required);
-		assert.strictEqual(inputNode.properties!.value, 'qux');
-
-		assert.strictEqual(vnode.properties!['form'], 'id2');
-		assert.strictEqual(labelNode.properties!.innerHTML, 'foo');
+		};
+
+		widget.setProperties(checkboxProperties);
+
+		const labelClasses = widget.classes(
+			css.root,
+			css.toggle,
+			css.checked,
+			css.disabled,
+			css.invalid,
+			css.readonly,
+			css.required
+		);
+
+		// TODO: is there something that works other than 'any'?
+		widget.expectRender(w<any>(Label, {
+			extraClasses: { root:
+				// TODO: any
+				parseLabelClasses(<any> labelClasses())
```

Comment:
Unfortunately, executing `labelClasses()` here seems to totally wreck the rest of my use of `widget.classes`, necessitating the manual creation of CSS class maps that you see in lines below.

Some help solving this would be appreciated.